src/snowflake/connector/cache.py

```diff
@@ -407,6 +412,34 @@ def __init__(
         if os.path.exists(self.file_path):
             self._load()
 
+    def _getitem(
+        self,
+        k: K,
+        *,
+        should_record_hits: bool = True,
+    ) -> V:
+        """non-locking version of __getitem__
+        This should only be used by internal functions when already
+        holding self._lock.
+        """
+        if k not in self._cache:
+            loaded = self._load_if_should()
+            if (not loaded) or k not in self._cache:
+                self._miss(k)
+                raise KeyError
+        t, v = self._cache[k]
+        if is_expired(t):
+            loaded = self._load_if_should()
+            expire_item = True
+            if loaded:
+                t, v = self._cache[k]
+                expire_item = is_expired(t)
+            if expire_item:
+                # Raises KeyError
+                self._expire(k)
+        self._hit(k)
+        return v
+
     def __getitem__(self, k: K) -> V:
```

Comment:
I think this function should reuse `self._getitem`, something like:
```python
 def __getitem__(self, k: K) -> V:
	with self._lock:
	    return self._getitem(k, v)
```

Reply:
I'm afraid we can't, if you remember, the `__getitem__` implementation of `SFDictFileCache` has more acquire/release operations in the middle and I think that's for performance consideration.